osu.Game/Graphics/UserInterface/ScreenTitle.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class ScreenTitle : CompositeDrawable, IHasAccentColour
+    {
+        private readonly SpriteIcon iconSprite;
+        private readonly OsuSpriteText titleText, pageText;
+
+        public FontAwesome Icon
+        {
+            get => iconSprite.Icon;
+            set => iconSprite.Icon = value;
+        }
+
+        public string Title
+        {
+            get => titleText.Text;
+            set => titleText.Text = value;
+        }
+
+        public string Page
+        {
+            get => pageText.Text;
+            set => pageText.Text = value;
+        }
+
+        public Color4 AccentColour
+        {
+            get => pageText.Colour;
+            set => pageText.Colour = value;
+        }
+
+        public ScreenTitle()
+        {
+            AutoSizeAxes = Axes.Both;
+
+            InternalChildren = new Drawable[]
+            {
+                iconSprite = new SpriteIcon
+                {
+                    Size = new Vector2(25),
+                    Anchor = Anchor.TopLeft,
+                    Origin = Anchor.TopRight,
+                    Margin = new MarginPadding { Right = 10 },
```

Comment:
Along with previous comment, I'd avoid this in-favor of using a secondary fillflow wrapping the whole thing with Spacing=10.